e2e/fiat_tf_test.go

```diff
@@ -0,0 +1,612 @@
+
+	unblacklistAccount(t, ctx, val, nw.fiatTfRoles.Blacklister, nw.fiatTfRoles.MasterMinter)
+
+	// ACTION: Configure an already configured Minter Controller with a new Minter
+	// EXPECTED: Success; Minter Controller is configured with Minter. The old minter should be disassociated
+	// from Minter Controller but keep its status and allowance
+	// Status:
+	// 	minterController1 -> minter1
+	// 	minterController2 -> minter2
+	//  minterController3 -> minter3
+
+	// configuring minter1 to ensure allowance stays the same after assigning minterController1 a new minter
+	_, err = val.ExecTx(ctx, minterController1.KeyName(), "fiat-tokenfactory", "configure-minter", minter1.FormattedAddress(), "1uusdc")
+	require.NoError(t, err, "error configuring minter controller")
+
+	showMinterPreUpdateMinterController, err := showMinters(ctx, val, minter1)
+	require.NoError(t, err, "failed to query show-minter")
+	expectedShowMinters := fiattokenfactorytypes.QueryGetMintersResponse{
+		Minters: fiattokenfactorytypes.Minters{
+			Address: minter1.FormattedAddress(),
+			Allowance: sdktypes.Coin{
+				Denom:  "uusdc",
+				Amount: math.OneInt(),
+			},
+		},
+	}
+	require.Equal(t, expectedShowMinters.Minters, showMinterPreUpdateMinterController.Minters, "configured minter and or allowance is not as expected")
+
+	w = interchaintest.GetAndFundTestUsers(t, ctx, "minter-4", math.OneInt(), noble)
+	minter4 := w[0]
+
+	_, err = val.ExecTx(ctx, nw.fiatTfRoles.MasterMinter.KeyName(), "fiat-tokenfactory", "configure-minter-controller", minterController1.FormattedAddress(), minter4.FormattedAddress())
+	require.NoError(t, err, "error configuring minter controller")
+
+	showMCRes, err = showMinterController(ctx, val, minterController1)
+	require.NoError(t, err, "failed to query show-minter-controller")
+	expectedShowMinterController = fiattokenfactorytypes.QueryGetMinterControllerResponse{
+		MinterController: fiattokenfactorytypes.MinterController{
+			Minter:     minter4.FormattedAddress(),
+			Controller: minterController1.FormattedAddress(),
+		},
+	}
+	require.Equal(t, expectedShowMinterController.MinterController, showMCRes.MinterController, "expected minter and minter controller is not as expected")
+
+	showMinterPostUpdateMinterController, err := showMinters(ctx, val, minter1)
+	require.NoError(t, err, "failed to query show-minter")
+	require.Equal(t, showMinterPreUpdateMinterController.Minters, showMinterPostUpdateMinterController.Minters, "the minter should not have changed since updating the minter controller with a new minter")
+
+	// ACTION:- Configure an already configured Minter to another Minter Controller
+	// EXPECTED: Success; Minter Controller is configured with new Minter. Minter can have multiple Minter Controllers.
+	// Status:
+	// 	minterController1 -> minter4
+	// 	minterController2 -> minter2
+	// 	minterController3 -> minter3
+	//  minter1 dissociated from minter but has allowance
```

Comment:
nit: reusing existing states in the same test makes the test more brittle. i.e. it would be better to create dedicated accounts for each test-case instead of reusing them.